### PR TITLE
[#IOCOM-566] added category field in ThirdPartyAttachment specification

### PR DESCRIPTION
**Pull request by acuiuli**
#### List of Changes

- [added category field in ThirdPartyAttachment specification](https://github.com/pagopa/io-backend/commit/854a81ba0ff33f5109a9ab940bc553f971a92986)

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by Garma00 — APPROVED** [review]

File: src/__mocks__/pn.ts [hunk 1]
```diff
@@ -52,12 +52,14 @@ export const aPNThirdPartyNotification = {
...
       name: "Atto",
       url: `/delivery/notifications/received/${aPnNotificationId}/attachments/documents/0`,
+      category: "DOCUMENT",
     },
     {
       id: "JGQG-YPJT-AUQZ-202301-R-1_DOC1",
       content_type: "application/pdf",
       name: "Lettera di accompagnamento",
       url: `/delivery/notifications/received/${aPnNotificationId}/attachments/documents/1`,
+      category: "DOCUMENT",
```
Comment (michaeldisaro) [inline src/__mocks__/pn.ts:62]: why not adding a test to check if providing no category results in a DEFAULT?
also, why not adding a test that shows that they can send an arbitrary category?

**Review by michaeldisaro — APPROVED** [review]

File: openapi/consumed/api-piattaforma-notifiche.yaml [hunk 2]
```diff
@@ -555,6 +555,9 @@ components:
         url:
           minLength: 1
           type: string
+        category:
+          type: string
```
Comment (michaeldisaro) [inline openapi/consumed/api-piattaforma-notifiche.yaml:559]: why not enforcing some constraints? min max length (3 - 30), a regex to expect just UPPERCASE_SNAKE_CASED_VALUES, with a right decoder to check this when it moves inside the backend?

**Review by michaeldisaro — APPROVED** [review]

**Pull request merged**